File: MessageUServer/base_response.py
class BaseResponse:
    def __init__(self, version, code, payload_size = 0):
        self.version = version
        self.code = code
        self.payload_size = payload_size


    def get_binary_response(self):
        return f"{self.version}{self.code}{self.payload_size}".encode()

File: MessageUClient/BaseResponse.cpp
#include "BaseResponse.h"

BaseResponse::BaseResponse(uint8_t version, uint16_t code, uint32_t payloadSize)
{
	this->version = version;
	this->code = code;
	this->payloadSize = payloadSize;
}

BaseResponse::~BaseResponse()
{
}

//getters
uint8_t BaseResponse::getVersion()
{
	return version;
}

uint16_t BaseResponse::getCode()
{
	return code;
}

uint32_t BaseResponse::getPayloadSize()
{
	return payloadSize;
}


